app/helper_test.go

```diff
@@ -105,11 +101,35 @@ func setupTestHelper(enterprise bool, tb testing.TB) *TestHelper {
 }
 
 func SetupEnterprise(tb testing.TB) *TestHelper {
-	return setupTestHelper(true, tb)
+	if testing.Short() {
+		tb.SkipNow()
+	}
+	dbStore := mainHelper.GetStore()
+	dbStore.DropAllTables()
+	dbStore.MarkSystemRanUnitTests()
+
+	return setupTestHelper(dbStore, true, tb)
 }
 
 func Setup(tb testing.TB) *TestHelper {
-	return setupTestHelper(false, tb)
+	if testing.Short() {
```

Comment:
If we follow the logic I mentioned above, then we don't need to split this into a separate UnitSetup function and just use the testing.Short() condition to pass the correct store to `setupTestHelper`. This would also reduce modification of a lot of lines which chance Setup to UnitSetup.

Reply:
This is not accidental, Setup is for setting the whole thing up, and UnitSetup is for setting things up but with a mock database. The former is going to be skipped entierly on `-short` flag, the later is going to be execute always. If we always use the mock for all tests depending on the `-short` flag, we need to implement all the mock actions to make it work properly.

This idea here is to keep tests without database needs, or where we don't need integrations tests quick, and keep the other integration tests working, with the less disruption possible.